flowcraft/templates/run_mlst.py

```diff
@@ -0,0 +1,431 @@
+    except (UnicodeDecodeError, AttributeError):
+        stderr = str(stderr)
+        stdout = str(stdout)
+
+    mlst_folder = os.path.abspath(os.path.realpath(stdout.splitlines()[0]))
+
+    mlst_db_path, species_scheme_map_new = get_species_scheme_map_version(mlst_folder)
+
+    scheme, genus_mlst_scheme = parse_species_scheme_map(species.lower().split(' '), mlst_db_path,
+                                                         species_scheme_map_new)
+
+    logger.info('MLST scheme found for {species}: {scheme}'.format(species=species, scheme=scheme))
+
+    species_genus = species.lower().split(' ')[0]
+
+    return scheme, species_genus, genus_mlst_scheme
+
+
+@MainWrapper
+def main(sample_id, assembly, expected_species):
+    """
+    Main executor of the mlst template.
+
+    Parameters
+    ----------
+    sample_id : str
+        Sample Identification string.
+    assembly : str
+        Fasta file.
+    expected_species : str
+        Expected species
+
+    """
+
+    pass_qc = False
+
+    novel_alleles = '{}_mlst_novel_alleles.fasta'.format(sample_id)
+
+    cli = [
+        'mlst',
+        '--novel',
+        novel_alleles,
+        assembly
+    ]
+
+    logger.debug("Running mlst subprocess with command: {}".format(cli))
+
+    p = subprocess.Popen(cli, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = p.communicate()
+
+    # Attempt to decode STDERR output from bytes. If unsuccessful, coerce to
+    # string
+    try:
+        stderr = stderr.decode("utf8")
+        stdout = stdout.decode("utf8")
+    except (UnicodeDecodeError, AttributeError):
+        stderr = str(stderr)
+        stdout = str(stdout)
+
+    logger.info("Finished mlst subprocess with STDOUT:\\n"
+                "======================================\\n{}".format(stdout))
+    logger.info("Fished mlst subprocess with STDERR:\\n"
+                "======================================\\n{}".format(stderr))
+    logger.info("Finished mlst with return code: {}".format(
+        p.returncode))
+
+    with open(".status", "w") as fh:
+        if p.returncode != 0:
+            fh.write("error")
+            return
+        else:
+            fh.write("pass")
+
+    if p.returncode == 0:
+
+        with open('{}.mlst.txt'.format(sample_id), 'wt') as writer:
+            writer.write(stdout)
+
+        # str
+        scheme_mlst = stdout.splitlines()[0].split('\\t')[1].split('_')[0]
+        # str
+        st = stdout.splitlines()[0].split('\\t')[2]
+        # list
+        profile = stdout.splitlines()[0].split('\\t')[3:]
+
+        # In case it's an unkown ST, cleans the novel alleles file.
+        if st == '-':
+            clean_novel_alleles(novel_alleles=novel_alleles, scheme_mlst=scheme_mlst, profile=profile)
+        else:
+            # in case mlst fails to create the novel alleles file
+            if os.path.isfile(novel_alleles):
+                os.remove(novel_alleles)
+
+        if not expected_species == "PASS":
```

Comment:
`PASS` could be something like `None`... given that it is used when species isn't provided. It could even be a default when the user doesn't provide the species.